### PR TITLE
Adds support for docker labels (fixes issue #50)

### LINKED ISSUE #50: Add support for docker labels
Add support for including image metadata in docker labels

### PR DESCRIPTION
**Pull request by rfrey-splunk**
Adds support to optionally include docker labels along with a simple example

### CODE REVIEW

File: roles/docker_image_build/vars/main.yml [hunk 1]
```diff
@@ -14,6 +14,8 @@ run_commands: "{{ role_run_commands + common_run_commands + host_run_commands }}
 
 build_variants: "{{ role_build_variants + common_build_variants + host_build_variants }}"
 
+docker_labels: "{{ role_docker_labels | combine(common_docker_labels) | combine(host_docker_labels) }}"
```
Comment (micahkemp-splunk) [inline roles/docker_image_build/vars/main.yml:17]: My vague recollection is that `combine` is for dictionaries.

```suggestion
docker_labels: "{{ role_docker_labels + common_docker_labels + host_docker_labels }}"
```

Comment (micahkemp-splunk) [inline roles/docker_image_build/vars/main.yml:17]: Thoughts on adding some `role_docker_labels`?  Perhaps at least build time, like we do for the [default indexed fields](https://cd.splunkdev.com/sas/ansible/playbooks/ansible_splunk_zero/-/blob/main/environments/docker_dev/group_vars/docker_build_images.yml#L135).

File: roles/docker_image_build/vars/main.yml [hunk 2]
```diff
@@ -94,3 +96,7 @@ role_run_commands:
   - command: "if [ -f {{ metricscheck_setup | quote }} ]; then python3 {{ metricscheck_setup | quote }} install; fi"
     user: root
     group: root
+
+role_docker_labels:
+  - name: org.opencontainers.image.base.name
+    value: "docker.io/library/ubuntu:{{ ubuntu_tag }}"
```
Comment (micahkemp-splunk) [inline roles/docker_image_build/vars/main.yml:102]: Why is this content in both `vars/` and `defaults/`?

Comment (rfrey-splunk) [reply roles/docker_image_build/vars/main.yml:102]: It's not

**Review by micahkemp-splunk — APPROVED** [review]
I realize this is still marked as a draft, but the content looks good so approving what's currently here.

**Review by micahkemp-splunk — APPROVED** [review]

**Review by midnight5ky — APPROVED** [review]

**Review by dtwersky — APPROVED** [review]

**Review by guohuang — APPROVED** [review]

**Pull request merged**